src/animation/Animation.test.ts

```diff
@@ -0,0 +1,81 @@
+import { Animation } from './Animation.js';
+
+
+class TestAnimation extends Animation {
+  public setUpCalled = false;
+  public tearDownCalled = false;
+  public animationCalls: Array<[number, number]> = [];
+  public beforeCalled = false;
+  public afterCalled = false;
+  
+  constructor() {
+    super(
+      (elapsedTime: number, deltaTime: number) => {
+        this.animationCalls.push([elapsedTime, deltaTime]);
+      },
+      { 
+        before: (() => { this.beforeCalled = true; }) as any,
+        after: (() => { this.afterCalled = true; }) as any
+      }
+    );
+  }
+
+  setUp(): void {
+    super.setUp();
+    this.setUpCalled = true;
+  }
+
+  tearDown(): void {
+    super.tearDown();
+    this.tearDownCalled = true;
+  }
+}
+
+// Test suite
+describe('Animation', () => {
+  let animation: TestAnimation;
+
+  beforeEach(() => {
+    animation = new TestAnimation();
+    animation.startTime = 1000;  // 1 second
```

Comment:
I prefer not to have comments unless something is especially confusing and instead structure the code so that the information in the comments is more obvious. In this case there's no need to say that they times are 1 second and 2 seconds.

Reply:
I also just noticed this but the times passed to the Animation class are in seconds, so these should just be 1, 2, etc.